tokio-udp/src/lib.rs

```diff
@@ -10,23 +10,28 @@
 //!
 //! The main struct for UDP is the [`UdpSocket`], which represents a UDP socket.
 //! Reading and writing to it can be done using futures, which return the
-//! [`RecvDgram`] and [`SendDgram`] structs respectively.
-//!
-//! For convenience it's also possible to convert raw datagrams into higher-level
-//! frames.
-//!
-//! [`UdpSocket`]: struct.UdpSocket.html
-//! [`RecvDgram`]: struct.RecvDgram.html
-//! [`SendDgram`]: struct.SendDgram.html
-//! [`UdpFramed`]: struct.UdpFramed.html
-//! [`framed`]: struct.UdpSocket.html#method.framed
+//! [`Recv`], [`Send`], [`RecvFrom`] and [`SendTo`] structs respectively.
 
-mod frame;
-mod recv_dgram;
-mod send_dgram;
+macro_rules! ready {
```

Comment:
this should be imported from `tokio-futures`

Reply:
This is TBD. Right now we have just been copy / pasting it around... I would just say keep it for now and we can fix later once we pick a strategy.

Reply:
Yeah, though I kinda wanted to avoid copying it everywhere 🤷‍♂ but im fine with keeping it.

Reply:
I mostly mean, uncopying can be handled in a follow up PR that does so across the repo.